### ISSUE TITLE
Framework dependency uses iterable return type which was added in PHP 7.1

### ISSUE THREAD

**Issue by webbower** [silverstripe/silverstripe-framework#8474]
## Affected Version

4.2.1

## Description

One of the dependencies `Symfony\Component\Filesystem\Filesystem::toIterable()` requires PHP 7.1 because it uses the `iterable` return type which was added in 7.1. This means SS 4.2.1 (at least) requires PHP 7.1.

https://github.com/symfony/filesystem/blob/v4.0.0/composer.json#L19
https://github.com/symfony/filesystem/blob/v4.0.0/Filesystem.php#L727
https://secure.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.types

## Steps to Reproduce

I found this while running `sake dev/build flush=1` on a VM upgrading from SilverStripe 3.6.4 to 4.2.1 and running PHP 7.0.x.

When it got to the `Index File_Versions` part, it failed:

```
// previous dev/build output omitted
  + Index File_Versions.ClassName: created as index ("ClassName")
  + Index File_Versions.ParentID: created as index ("ParentID")
  + Index File_Versions.OwnerID: created as index ("OwnerID")
ERROR [Emergency]: Uncaught TypeError: Return value of Symfony\Component\Filesystem\Filesystem::toIterable() must be an instance of Symfony\Component\Filesystem\iterable, array returned
IN GET dev/build
Line 729 in /var/www/html/vendor/symfony/filesystem/Filesystem.php

Source
======
  720:         }
  721: 
  722:         if (false === @file_put_contents($filename, $content, FILE_APPEND)) {
  723:             throw new IOException(sprintf('Failed to write file "%s".', $filename), 0, null,
       $filename);
  724:         }
  725:     }
  726: 
  727:     private function toIterable($files): iterable
  728:     {
* 729:         return \is_array($files) || $files instanceof \Traversable ? $files : array($files);
  730:     }
  731: 
  732:     /**
  733:      * Gets a 2-tuple of scheme (may be null) and hierarchical part of a filename (e.g. file:///tmp
       -> array(file, tmp)).
  734:      */
  735:     private function getSchemeAndHierarchy(string $filename): array

Trace
=====
Symfony\Component\Filesystem\Filesystem->toIterable(/var/www/html/silverstripe-cache/vagrant/catalogue.en_US.1cd7e874b24ab41081c7781e4161053bf515fc91.php)
Filesystem.php:201

Symfony\Component\Filesystem\Filesystem->chmod(/var/www/html/silverstripe-cache/vagrant/catalogue.en_US.1cd7e874b24ab41081c7781e4161053bf515fc91.php, 438, 2)
ResourceCheckerConfigCache.php:126

Symfony\Component\Config\ResourceCheckerConfigCache->write(<?php

use Symfony\Component\Translation\MessageCatalogue;

$catalogue = new MessageCatalogue('en_US', array (
  'messages' => 
  array (
  ),
));

$catalogueEn = new MessageCatalogue('en', array (
  'messages' => 
  array (
    'AssetFormFactory.php.SilverStripe\\AssetAdmin\\Forms\\AssetFormFactorySAVE' => 'Save',
    'AssetFormFactory.php.SilverStripe\\AssetAdmin\\Forms\\AssetFormFactorySAVED' => 'Saved',
    'CampaignAdmin.php.SilverStripe\\CampaignAdmin\\CampaignAdminSAVE' => 'Save',
    'CampaignAdmin.php.SilverStripe\\CampaignAdmin\\CampaignAdminSAVED' => 'Saved',
    'FileFormFactory.php.SilverStripe\\AssetAdmin\\Forms\\FileFormFactoryPUBLISH' => 'Publish',
    'FileFormFactory.php.SilverStripe\\AssetAdmin\\Forms\\FileFormFactoryPUBLISHED' => 'Published',
    'MoveFormFactory.php.SilverStripe\\AssetAdmin\\Forms\\MoveFormFactoryCHOOSEFOLDER' => 'Choose a folder',
    'MoveFormFactory.php.SilverStripe\\AssetAdmin\\Forms\\MoveFormFactoryMOVE' => 'Move',
    'SilverStripe\\Admin\\CMSProfileController.CANTEDIT' => 'You don\'t have permission to do that',
    'SilverStripe\\Admin\\CMSProfileController.MENUTITLE' => 'My Profile',
    'SilverStripe\\Admin\\Forms\\EditorEmailLinkFormFactory.EMAIL' => 'Email address',
    'SilverStripe\\Admin\\Forms\\EditorEmailLinkFormFactory.LINKDESCR' => 'Link description',
    'SilverStripe\\Admin\\Forms\\EditorEmailLinkFormFactory.LINKTEXT' => 'Link text',
    'SilverStripe\\Admin\\Forms\\EditorEmailLinkFormFactory.SUBJECT' => 'Subject',
    'SilverStripe\\Admin\\Forms\\EditorExternalLinkFormFactory.ANCHORVALUE' => 'Anchor',
    'SilverStripe\\Admin\\Forms\\EditorExternalLinkFormFactory.LINKDESCR' => 'Link description',
    'SilverStripe\\Admin\\Forms\\EditorExternalLinkFormFactory.LINKOPENNEWWIN' => 'Open in new window/tab',
    'SilverStripe\\Admin\\Forms\\EditorExternalLinkFormFactory.LINKTEXT' => 'Link text',
    'SilverStripe\\Admin\\Forms\\EditorExternalLinkFormFactory.URL' => 'URL',
    'SilverStripe\\Admin\\Forms\\LinkFormFactory.INSERT_LINK' => 'Insert link',
    'SilverStripe\\Admin\\Forms\\UsedOnTable.ADDEDTODRAFT' => 'Draft',
    'SilverStripe\\Admin\\Forms\\UsedOnTable.MODIFIEDONDRAFTSHORT' => 'Modified',
    'SilverStripe\\Admin\\GroupImportForm.Help1' => '<p>Import one or more groups in <em>CSV</em> format (comma-separated values). <small><a href="#" class="toggle-advanced">Show advanced usage</a></small></p>',
    'SilverStripe\\Admin\\GroupImportForm.Help2' => '<div class="advanced"><h4>Advanced usage</h4><ul><li>Allowed columns: <em>{columns}</em></li><li>Existing groups are matched by their unique <em>Code</em> value, and updated with any new values from the imported file</li><li>Group hierarchies can be created by using a <em>ParentCode</em> column.</li><li>Permission codes can be assigned by the <em>PermissionCode</em> column. Existing permission codes are not cleared.</li></ul></div>',
    'SilverStripe\\Admin\\GroupImportForm.ResultCreated' => 'Created {count} groups',
    'SilverStripe\\Admin\\GroupImportForm.ResultDeleted' => 'Deleted {count} groups',
    'SilverStripe\\Admin\\GroupImportForm.ResultUpdated' => 'Updated {count} groups',
    'SilverStripe\\Admin\\LeftAndMain.ACCESSALLINTERFACES' => 'Access to all CMS sections',
    'SilverStripe\\Admin\\LeftAndMain.ACCESSALLINTERFACESHELP' => 'Overrules more specific access settings.',
    'SilverStripe\\Admin\\LeftAndMain.BROWSER_WARNING_HEADING' => 'Unsupported browser',
    'SilverStripe\\Admin\\LeftAndMain.BROWSER_WARNING_TEXT' => 'You&apos;re using a web browser that&apos;s not supported by SilverStripe.<br />Please upgrade to one of the following options.',
    'SilverStripe\\Admin\\LeftAndMain.CollapsePanel' => 'Collapse Panel',
    'SilverStripe\\Admin\\LeftAndMain.DELETED' => 'Deleted.',
    'SilverStripe\\Admin\\LeftAndMain.DropdownBatchActionsDefault' => 'Choose an action...',
    'SilverStripe\\Admin\\LeftAndMain.ExpandPanel' => 'Expand Panel',
    'SilverStripe\\Admin\\LeftAndMain.HELP' => 'Help',
    'SilverStripe\\Admin\\LeftAndMain.LOGOUT' => 'Log out',
    'SilverStripe\\Admin\\LeftAndMain.MenuToggleAuto' => 'Auto',
    'SilverStripe\\Admin\\LeftAndMain.MenuToggleStickyNav' => 'Sticky nav',
    'SilverStripe\\Admin\\LeftAndMain.NavigateUp' => 'Navigate up a folder',
    'SilverStripe\\Admin\\LeftAndMain.PERMAGAIN' => 'You have been logged out of the CMS.  If you would like to log in again, enter a username and password below.',
    'SilverStripe\\Admin\\LeftAndMain.PERMALREADY' => 'I\'m sorry, but you can\'t access that part of the CMS.  If you want to log in as someone else, do so below.',
    'SilverStripe\\Admin\\LeftAndMain.PERMDEFAULT' => 'You must be logged in to access the administration area; please enter your credentials below.',
    'SilverStripe\\Admin\\LeftAndMain.PreviewButton' => 'Preview',
    'SilverStripe\\Admin\\LeftAndMain.REQUIREJS' => 'The CMS requires that you have JavaScript enabled.',
    'SilverStripe\\Admin\\LeftAndMain.SAVEDUP' => 'Saved.',
    'SilverStripe\\Admin\\LeftAndMain.SUBMIT_BUTTON_LABEL' => 'Go',
    'SilverStripe\\Admin\\LeftAndMain.ValidationError' => 'Validation error',
    'SilverStripe\\Admin\\LeftAndMain.VersionUnknown' => 'unknown',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.Auto' => 'Auto',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.ChangeViewMode' => 'Change view mode',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.Desktop' => 'Desktop',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.DualWindowView' => 'Dual Window',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.EditView' => 'Edit mode',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.Mobile' => 'Mobile',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.PreviewState' => 'Preview State',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.PreviewView' => 'Preview mode',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.Responsive' => 'Responsive',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.SplitView' => 'Split mode',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.Tablet' => 'Tablet',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.ViewDeviceWidth' => 'Select a preview width',
    'SilverStripe\\Admin\\LeftAndMain_ViewModeSelector.Width' => 'width',
    'SilverStripe\\Admin\\MemberImportForm.BtnImport' => 'Import from CSV',
    'SilverStripe\\Admin\\MemberImportForm.FileFieldLabel' => 'CSV File <small>(Allowed extensions: *.csv)</small>',
    'SilverStripe\\Admin\\MemberImportForm.Help1' => '<p>Import users in <em>CSV format</em> (comma-separated values). <small><a href="#" class="toggle-advanced">Show advanced usage</a></small></p>',
    'SilverStripe\\Admin\\MemberImportForm.Help2' => '<div class="advanced"><h4>Advanced usage</h4><ul><li>Allowed columns: <em>{columns}</em></li><li>Existing users are matched by their unique <em>Code</em> property, and updated with any new values from the imported file.</li><li>Groups can be assigned by the <em>Groups</em> column. Groups are identified by their <em>Code</em> property, multiple groups can be separated by comma. Existing group memberships are not cleared.</li></ul></div>',
    'SilverStripe\\Admin\\MemberImportForm.ResultCreated' => 'Created {count} members',
    'SilverStripe\\Admin\\MemberImportForm.ResultDeleted' => 'Deleted {count} members',
    'SilverStripe\\Admin\\MemberImportForm.ResultNone' => 'No changes',
    'SilverStripe\\Admin\\MemberImportForm.ResultUpdated' => 'Updated {count} members',
    'SilverStripe\\Admin\\ModelAdmin.APPLY_FILTER' => 'Apply Filter',
    'SilverStripe\\Admin\\ModelAdmin.CLEARSEARCH' => 'Clear search',
    'SilverStripe\\Admin\\ModelAdmin.DELETE' => 'Delete',
    'SilverStripe\\Admin\\ModelAdmin.DELETEDRECORDS' => 'Deleted {count} records.',
    'SilverStripe\\Admin\\ModelAdmin.EMPTYBEFOREIMPORT' => 'Replace data',
    'SilverStripe\\Admin\\ModelAdmin.IMPORT' => 'Import from CSV',
    'SilverStripe\\Admin\\ModelAdmin.IMPORTEDRECORDS' => 'Imported {count} records.',
    'SilverStripe\\Admin\\ModelAdmin.IMPORTSPECFIELDS' => 'Database columns',
    'SilverStripe\\Admin\\ModelAdmin.IMPORTSPECLINK' => 'Show Specification for {model}',
    'SilverStripe\\Admin\\ModelAdmin.IMPORTSPECRELATIONS' => 'Relations',
    'SilverStripe\\Admin\\ModelAdmin.IMPORTSPECTITLE' => 'Specification for {model}',
    'SilverStripe\\Admin\\ModelAdmin.NOCSVFILE' => 'Please browse for a CSV file to import',
    'SilverStripe\\Admin\\ModelAdmin.NOIMPORT' => 'Nothing to import',
    'SilverStripe\\Admin\\ModelAdmin.RESET' => 'Reset',
    'SilverStripe\\Admin\\ModelAdmin.SEARCH' => 'Search',
    'SilverStripe\\Admin\\ModelAdmin.SEARCHRESULTSFOR' => 'Search results for',
    'SilverStripe\\Admin\\ModelAdmin.Title' => 'Data Models',
    'SilverStripe\\Admin\\ModelAdmin.UPDATEDRECORDS' => 'Updated {count} records.',
    'SilverStripe\\Admin\\SecurityAdmin.ACCESS_HELP' => 'Allow viewing, adding and editing users, as well as assigning permissions and roles to them.',
    'SilverStripe\\Admin\\SecurityAdmin.APPLY_ROLES' => 'Apply roles to groups',
    'SilverStripe\\Admin\\SecurityAdmin.APPLY_ROLES_HELP' => 'Ability to edit the roles assigned to a group. Requires the "Access to \'Users\' section" permission.',
    'SilverStripe\\Admin\\SecurityAdmin.EDITPERMISSIONS' => 'Manage permissions for groups',
    'SilverStripe\\Admin\\SecurityAdmin.EDITPERMISSIONS_HELP' => 'Ability to edit Permissions and IP Addresses for a group. Requires the "Access to \'Security\' section" permission.',
    'SilverStripe\\Admin\\SecurityAdmin.IMPORTGROUPS' => 'Import groups',
    'SilverStripe\\Admin\\SecurityAdmin.IMPORTUSERS' => 'Import users',
    'SilverStripe\\Admin\\SecurityAdmin.MENUTITLE' => 'Security',
    'SilverStripe\\Admin\\SecurityAdmin.MemberListCaution' => 'Caution: Removing members from this list will remove them from all groups and the database',
    'SilverStripe\\Admin\\SecurityAdmin.TABROLES' => 'Roles',
    'SilverStripe\\Admin\\SecurityAdmin.Users..., Array)
Translator.php:360

Symfony\Component\Translation\Translator->dumpCatalogue(en_US, Symfony\Component\Config\ResourceCheckerConfigCache)
Translator.php:320

Symfony\Component\Translation\Translator->Symfony\Component\Translation\{closure}(Symfony\Component\Config\ResourceCheckerConfigCache)
ResourceCheckerConfigCacheFactory.php:43

Symfony\Component\Config\ResourceCheckerConfigCacheFactory->cache(/var/www/html/silverstripe-cache/vagrant/catalogue.en_US.1cd7e874b24ab41081c7781e4161053bf515fc91.php, Closure)
Translator.php:321

Symfony\Component\Translation\Translator->initializeCacheCatalogue(en_US)
Translator.php:285

Symfony\Component\Translation\Translator->loadCatalogue(en_US)
Translator.php:239

Symfony\Component\Translation\Translator->getCatalogue(en_US)
Translator.php:200

Symfony\Component\Translation\Translator->trans(SilverStripe\SiteConfig\SiteConfig.SITENAMEDEFAULT, Array, messages, en_US)
SymfonyMessageProvider.php:97

SilverStripe\i18n\Messages\Symfony\SymfonyMessageProvider->translate(SilverStripe\SiteConfig\SiteConfig.SITENAMEDEFAULT, Your Site Name, Array)
i18n.php:206

SilverStripe\i18n\i18n::_t(SilverStripe\SiteConfig\SiteConfig.SITENAMEDEFAULT, Your Site Name)
functions.php:63

_t(SilverStripe\SiteConfig\SiteConfig.SITENAMEDEFAULT, Your Site Name)
SiteConfig.php:75

SilverStripe\SiteConfig\SiteConfig->populateDefaults()
DataObject.php:383

SilverStripe\ORM\DataObject->__construct()

ReflectionClass->newInstance()
InjectionCreator.php:26

SilverStripe\Core\Injector\InjectionCreator->create(SilverStripe\SiteConfig\SiteConfig, Array)
Injector.php:585

SilverStripe\Core\Injector\Injector->instantiate(Array, SilverStripe\SiteConfig\SiteConfig, singleton)
Injector.php:993

SilverStripe\Core\Injector\Injector->getNamedService(SilverStripe\SiteConfig\SiteConfig, 1, Array)
Injector.php:946

SilverStripe\Core\Injector\Injector->get(SilverStripe\SiteConfig\SiteConfig)
functions.php:34

singleton(SilverStripe\SiteConfig\SiteConfig)
DatabaseAdmin.php:294

SilverStripe\ORM\DatabaseAdmin->SilverStripe\ORM\{closure}()
DBSchemaManager.php:161

SilverStripe\ORM\Connect\DBSchemaManager->schemaUpdate(Closure)
DatabaseAdmin.php:323

SilverStripe\ORM\DatabaseAdmin->doBuild(, 1)
DatabaseAdmin.php:147

SilverStripe\ORM\DatabaseAdmin->build()
DatabaseAdmin.php:119

SilverStripe\ORM\DatabaseAdmin->index(SilverStripe\Control\HTTPRequest)
RequestHandler.php:320

SilverStripe\Control\RequestHandler->handleAction(SilverStripe\Control\HTTPRequest, index)
Controller.php:284

SilverStripe\Control\Controller->handleAction(SilverStripe\Control\HTTPRequest, index)
RequestHandler.php:202

SilverStripe\Control\RequestHandler->handleRequest(SilverStripe\Control\HTTPRequest)
Controller.php:212

SilverStripe\Control\Controller->handleRequest(SilverStripe\Control\HTTPRequest)
DevBuildController.php:24

SilverStripe\Dev\DevBuildController->build(SilverStripe\Control\HTTPRequest)
RequestHandler.php:320

SilverStripe\Control\RequestHandler->handleAction(SilverStripe\Control\HTTPRequest, build)
Controller.php:284

SilverStripe\Control\Controller->handleAction(SilverStripe\Control\HTTPRequest, build)
RequestHandler.php:202

SilverStripe\Control\RequestHandler->handleRequest(SilverStripe\Control\HTTPRequest)
Controller.php:212

SilverStripe\Control\Controller->handleRequest(SilverStripe\Control\HTTPRequest)
RequestHandler.php:226

SilverStripe\Control\RequestHandler->handleRequest(SilverStripe\Control\HTTPRequest)
Controller.php:212

SilverStripe\Control\Controller->handleRequest(SilverStripe\Control\HTTPRequest)
Director.php:361

SilverStripe\Control\Director->SilverStripe\Control\{closure}(SilverStripe\Control\HTTPRequest)
VersionedHTTPMiddleware.php:40

SilverStripe\Versioned\VersionedHTTPMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
BasicAuthMiddleware.php:68

SilverStripe\Security\BasicAuthMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
AuthenticationMiddleware.php:61

SilverStripe\Security\AuthenticationMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
CanonicalURLMiddleware.php:188

SilverStripe\Control\Middleware\CanonicalURLMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
HTTPCacheControlMiddleware.php:42

SilverStripe\Control\Middleware\HTTPCacheControlMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
ChangeDetectionMiddleware.php:27

SilverStripe\Control\Middleware\ChangeDetectionMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
FlushMiddleware.php:29

SilverStripe\Control\Middleware\FlushMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
RequestProcessor.php:66

SilverStripe\Control\RequestProcessor->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
SessionMiddleware.php:20

SilverStripe\Control\Middleware\SessionMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
AllowedHostsMiddleware.php:60

SilverStripe\Control\Middleware\AllowedHostsMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
TrustedProxyMiddleware.php:176

SilverStripe\Control\Middleware\TrustedProxyMiddleware->process(SilverStripe\Control\HTTPRequest, Closure)
HTTPMiddlewareAware.php:62

SilverStripe\Control\Director->SilverStripe\Control\Middleware\{closure}(SilverStripe\Control\HTTPRequest)
HTTPMiddlewareAware.php:65

SilverStripe\Control\Director->callMiddleware(SilverStripe\Control\HTTPRequest, Closure)
Director.php:370

SilverStripe\Control\Director->handleRequest(SilverStripe\Control\HTTPRequest)
HTTPApplication.php:48

SilverStripe\Control\HTTPApplication->SilverStripe\Control\{closure}(SilverStripe\Control\HTTPRequest)

call_user_func(Closure, SilverStripe\Control\HTTPRequest)
HTTPApplication.php:66

SilverStripe\Control\HTTPApplication->SilverStripe\Control\{closure}(SilverStripe\Control\HTTPRequest)
HTTPMiddlewareAware.php:65

SilverStripe\Control\HTTPApplication->callMiddleware(SilverStripe\Control\HTTPRequest, Closure)
HTTPApplication.php:67

SilverStripe\Control\HTTPApplication->execute(SilverStripe\Control\HTTPRequest, Closure, 1)
HTTPApplication.php:49

SilverStripe\Control\HTTPApplication->handle(SilverStripe\Control\HTTPRequest)
cli-script.php:22
```

**Comment by kinglozzer** [comment]
This is often caused by having a different PHP version on the command line, if you run `php -v` which version do you see?

**Comment by webbower** [comment]
It occurs to me that since `silverstripe/upgrader` requires 7.1 while SilverStripe itself requires >= 5.6, this bug could have flown under the radar since anyone upgrading from 3 -> 4 and using the upgrader will have 7.1 installed. I upgraded PHP on my host machine to 7.1 in order to use the upgrader. But I'm using a Vagrant VM (using https://box.scotch.io v3.5) to run the site. `php -v` from inside the VM is:

```
PHP 7.0.18-0ubuntu0.16.04.1 (cli) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.18-0ubuntu0.16.04.1, Copyright (c) 1999-2017, by Zend Technologies
```

**Comment by webbower** [comment]
It looks like the previous version of `symfony/filesystem` (3.4.17) doesn't require 7.1 https://github.com/symfony/filesystem/blob/v3.4.17/composer.json#L19

I might try doing a forced downgrade to that version to start with.

**Comment by sminnee** [comment]
@webbower are you running different versions of PHP on your dev and production systems?

If so, that's generally not a great idea, but if you have to do that for some reason, you can edit your composer.json to pin the PHP version to install dependencies for. Here's a post about it: https://andy-carter.com/blog/composer-php-platform

I'm not sure that there's any bug here. Do you want to try that fix, and if it works for you, close this issue?

**Comment by webbower** [comment]
Nope. Running PHP 7.0.x consistently across all web code-running environments because that's what our DevOps team currently supports. I only upgraded my host machine to 7.1 since `silverstripe/upgrader` requires it. Running SS in a VirtualBox VM with PHP 7.0 (see above output) but did the upgrader commands on my host Mac with 7.1. My host machine doesn't run any of the web code for our site for local dev.

UPDATE: I dug into the dependency tree for `symphony/filesystem` and found a functional workaround. `silverstripe-framework` requires `symphony/config` which has the dependency on `symphony/filesystem` version(s) `"~2.8|~3.0|~4.0"`. So I explicitly set the `symphony/filesystem` version in my top-level `require` block to `^3.4` which installed the latest version of 3.x, which doesn't have the PHP 7.1 dependency (PHP dep is `^5.5.9|>=7.0.8` for that version). You might want to add that rule to the `composer.json` for `silverstripe-framework` until you bump up PHP requirement for the Framework to 7.1

**Comment by sminnee** [comment]
You need to put this in your project’s composer.json

```
“config": {
        "platform": {
            "php": "7.0.0"
        }
    },
```

That should fix this issue for you.

There’s no core fix needed for this.

**Comment by robbieaverill** [comment]
You can also run `composer update` from your project root, which will install dependencies such as this that are compatible with your PHP version. If you've installed SilverStripe from an archive (e.g. from the website's download section) it may come with dependencies for PHP 7.1, but you can easily adjust this. Thanks for raising the issue, I agree with @sminnee that there's no core fix here, so I'm going to close the issue

**Comment by kinglozzer** [comment]
> Running SS in a VirtualBox VM with PHP 7.0 (see above output) but did the upgrader commands on my host Mac with 7.1

That is almost certainly the cause. I believe the upgrader will update composer dependencies, and if it’s running as PHP 7.1 it’ll pull in 7.1-compatible dependencies. You should be able to use either @sminnee’s workaround above, or if you use @robbieaverill’s just make sure you run `composer update` from inside your VM and you should be good to go 👍

Sounds like the same issue described in https://github.com/silverstripe/silverstripe-upgrader/issues/115, so it’s probably best to track this there as it definitely sounds more like an issue with the upgrader than framework itself.

**Comment by webbower** [comment]
I did finally figure out the root cause. I ran `composer update` from my host (running PHP 7.1) and so it used that for PHP version dependencies. When I re-ran it from the VM, it installed the correct versions of everything because it was running in a PHP 7.0 environment. SO agreed with @robbieaverill that this is a non-issue.

@sminnee I might just add that configuration to be on the safe side. Thanks for that.
